test/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotAddOrRemoveStaticKeywordTests.cs

```diff
@@ -0,0 +1,282 @@
+        }
+
+        [Fact]
+        public static void ReproductionCaseFromIssue()
+        {
+            // This is the reproduction case from the issue
+            string leftSyntax = @"
+namespace CompatTests
+{
+    public class Class1
+    {
+        public static string Foo => ""net8.0"";
+        public string Bar => ""net8.0"";
+    }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+    public class Class1
+    {
+        public string Foo => ""net10.0"";
+        public static string Bar => ""net10.0"";
+    }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new(s_ruleFactory);
+
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            // We expect 4 differences: 2 for Foo (property + getter), 2 for Bar (property + getter)
+            CompatDifference[] expected = new[]
+            {
+                CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotRemoveStaticFromMember, string.Empty, DifferenceType.Removed, "P:CompatTests.Class1.Foo"),
+                CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotRemoveStaticFromMember, string.Empty, DifferenceType.Removed, "M:CompatTests.Class1.get_Foo"),
+                CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotAddStaticToMember, string.Empty, DifferenceType.Added, "P:CompatTests.Class1.Bar"),
+                CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotAddStaticToMember, string.Empty, DifferenceType.Added, "M:CompatTests.Class1.get_Bar"),
+            };
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void RemoveStaticFromTypeInStrictModeReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+    public static class First
+    {
+        public static void F() {}
+    }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+    public class First
+    {
+        public static void F() {}
+    }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new(s_ruleFactory, new ApiComparerSettings(strictMode: true));
+
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(left, right);
+
+            CompatDifference[] expected = new[]
+            {
+                CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotRemoveStaticFromType, string.Empty, DifferenceType.Removed, "T:CompatTests.First"),
+            };
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void RemoveStaticFromTypeNotReportedInNonStrictMode()
```

Comment:
Can we combine this with RemoveStaticFromTypeInStrictModeReported and make it a theory over true/false for strict mode?